playerbot/strategy/actions/ReachTargetActions.h

```diff
@@ -10,43 +10,44 @@ namespace ai
     class ReachTargetAction : public MovementAction
     {
     public:
-        ReachTargetAction(PlayerbotAI* ai, string name, float distance) : MovementAction(ai, name)
-		{
-            this->distance = distance;
-        }
+        ReachTargetAction(PlayerbotAI* ai, string name, float distance) : MovementAction(ai, name), distance(distance) {}
+
         virtual bool Execute(Event& event)
 		{
             Unit* target = AI_VALUE(Unit*, GetTargetName());
-            if (!target)
-                return false;
+            if (target)
+            {
+                UpdateMovementState();
 
-            UpdateMovementState();
+                // Ignore movement if too far
+                const float distanceToTarget = sServerFacade.GetDistance2d(bot, target);
+                if (distanceToTarget <= sPlayerbotAIConfig.sightDistance)
+                {
+                    float chaseDist = distance;
+                    const bool inLos = bot->IsWithinLOSInMap(target, true);
+                    const bool isFriend = sServerFacade.IsFriendlyTo(bot, target);
 
-            float combatReach = bot->GetCombinedCombatReach(target, true);
+                    if (distance > 0.0f)
+                    {
+                        const float coeff = isFriend ? 0.8f : 1.0f;
+                        chaseDist = inLos ? distance : (isFriend ? std::min(distanceToTarget * 0.9f, distance) : distance);
+                        chaseDist = (chaseDist - sPlayerbotAIConfig.contactDistance) * coeff;
+                    }
 
-            if (distance == ATTACK_DISTANCE)
-            {
-                return ChaseTo(target, 0.0f, bot->GetAngle(target));
-            }
-            else
-            {
-                float distanceToTarget = sServerFacade.GetDistance2d(bot, target);
-                if (distanceToTarget > sPlayerbotAIConfig.sightDistance)
-                    return false;
-
-                combatReach = bot->GetCombinedCombatReach(target, false);
-                bool inLos = bot->IsWithinLOSInMap(target, true);
-                bool isFriend = sServerFacade.IsFriendlyTo(bot, target);
-                float chaseDist = inLos ? distance : isFriend ? std::min(distanceToTarget * 0.9f, distance) : distance;
-                float coeff = 0.8f;
-                if (!isFriend)
-                    coeff = 1.0f;
-                if (inLos || isFriend)
-                    return MoveNear(target, chaseDist);
-
-                return ChaseTo(target, (chaseDist - sPlayerbotAIConfig.contactDistance) * coeff, bot->GetAngle(target));
+                    if (inLos && isFriend && (distance <= sPlayerbotAIConfig.followDistance))
+                    {
+                        return MoveNear(target, chaseDist);

```

Comment:
MoveNear does ignore the distance if it is higher than the default follow distance. Not sure what the intention was here but for most cases ChaseTo will be used instead